### PR TITLE
Feat/user sign up

### LINKED ISSUE #13: Allow users to sign in/up.
We need a way to allow users to sign up. Ideally they could do that using their email + password or using oauth2.

## Tasks to complete

- [ ] Create a sign up page
- [ ] Create a sign in page
- [ ] Add some oauth providers to those pages.

## Acceptance criteria

- [ ] Users can sign up using email+password
- [ ] Users can sign up using oauth.

### PR DESCRIPTION
**Pull request by DanielRailean**
Closes: https://github.com/SEP6-A2022/SEP6/issues/13

### CODE REVIEW

**Comment by DanielRailean** [comment]
Seems like email + password auth is still pending

**Comment by DanielRailean** [comment]
Made registration/login using email + pass work.

**Comment by DanielRailean** [comment]
@asterixcode have a look.

File: db_load/load_db.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import fs from 'fs';
+import * as child_process from "child_process";
+import * as util from "util";
+const exec = util.promisify(child_process.exec);
+
+// export pg pass 
+// export PGPASSWORD=
+
+const main =async () => {
+    const testFolder = './out/';
```
Comment (asterixcode) [inline db_load/load_db.ts:10]: maybe add the ./out/ to the .gitignore?

File: product/client/src/routes/user/+page.svelte [hunk 2]
```diff
@@ -0,0 +1,81 @@
+<script lang="ts">
+	import { goto } from "$app/navigation";
+	import { onMount } from "svelte";
+
+    let email = ""
+    let password = ""
+    let password2 = ""
+
+    let clientid = "11c2d39160e19c5d2ab0"
```
Comment (asterixcode) [inline product/client/src/routes/user/+page.svelte:9]: maybe have the clientid as secrets / env variable?

File: product/client/src/routes/user/auth/+page.server.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    const secret = process.env.GITHUB_OAUTH_SECRET
+    const client_id = process.env.GITHUB_CLIENT_ID
+    const accessSigningKey = process.env.JWT_ACCESS_TOKEN_SECRET_STRING
+
+    if(!secret) throw error(500, "Internal error. Code 1")
+    if(!client_id) throw error(500, "Internal error. Code 2")
+    if(!code) throw error(404, "Code not found in the request! Code 3");
+    if(!accessSigningKey) throw error(500, "Internal error! Code 4");
+
+    const admins = ["me@ddlele.com"]
```
Comment (asterixcode) [inline product/client/src/routes/user/auth/+page.server.ts:22]: secrets?

File: product/infra/src/docker.ts [hunk 4]
```diff
@@ -7,6 +7,20 @@ export const deployContainer = (config: IDockerSettings, dbConnectionString: str
     // Location to deploy Cloud Run services
     const location = gcp.config.zone ?? ""
 
+    const githubClientId = "11c2d39160e19c5d2ab0"
```
Comment (asterixcode) [inline product/infra/src/docker.ts:12]: maybe have it as secrets? or not needed?

File: db_load/load_db.ts (hunk 1 above)
Comment (asterixcode) [inline db_load/load_db.ts:10]: maybe add the ./out/ to the .gitignore?
and remove the folder here...

File: product/client/src/routes/user/+page.svelte (hunk 2 above)
Comment (DanielRailean) [reply product/client/src/routes/user/+page.svelte:9]: The client it is not really a secret
Its used in the url for the user.

File: product/infra/src/docker.ts (hunk 4 above)
Comment (DanielRailean) [reply product/infra/src/docker.ts:12]: same as above

File: product/client/src/routes/user/auth/+page.server.ts (hunk 3 above)
Comment (DanielRailean) [reply product/client/src/routes/user/auth/+page.server.ts:22]: Not sure what is secret in my email :D

Comment (DanielRailean) [reply product/client/src/routes/user/auth/+page.server.ts:22]: Maybe it shouldn't be hardcoded, but should be fine for now.

File: db_load/load_db.ts (hunk 1 above)
Comment (DanielRailean) [reply db_load/load_db.ts:10]: will remove the /out and its files.

File: product/client/src/routes/user/auth/+page.server.ts (hunk 3 above)
Comment (asterixcode) [reply product/client/src/routes/user/auth/+page.server.ts:22]: yeah, just thought that maybe you didnt want to have your email hardcoded / exposed here...
it's up to you :)

**Review by asterixcode — APPROVED** [review]

**Pull request merged**